src/main/src/main/java/org/geoserver/catalog/impl/DefaultCatalogFacade.java

```diff
@@ -200,10 +200,17 @@ public void remove(StoreInfo store) {
     }
     
     public void save(StoreInfo store) {
-        beforeSaved(store);
+        ModificationProxy h = (ModificationProxy) Proxy.getInvocationHandler(store);
```

Comment:
I see this patter repeated over and over and over... how about a method taking a lambda to do the bit in the middle, after the notication and before commitProxy? Just thinking out loud.